src/components/molecules/HistoricalChart/HistoricalChart.tsx

```diff
@@ -0,0 +1,353 @@
+            createCandlestickChart(chartContainerRef.current);
+
+        const { volumeSeries, chart: volumeChart } = createVolumeChart(
+            secondContainerRef.current,
+            timeScale
+        );
+
+        setupCharts(candlestickSeries, volumeSeries);
+
+        subscribeToChartEvents(
+            candleStickChart,
+            volumeChart,
+            candlestickSeries,
+            volumeSeries
+        );
+
+        volumeChart.timeScale();
+        candleStickChart.timeScale();
+
+        const updateTitleData = (
+            param: MouseEventParams<Time>,
+            sourceChart: string
+        ) => {
+            if (!param.seriesData.size) return;
+
+            let index = 0;
+            if (param.time) {
+                index = candlestickSeries
+                    .data()
+                    .findIndex(
+                        (item: Record<'time', Time>) => item.time === param.time
+                    );
+            }
+
+            const candleData = candlestickSeries.dataByIndex(index);
+            const volumeData = volumeSeries.dataByIndex(index);
+            if (sourceChart === 'candlestick') {
+                syncCrosshair(volumeChart, volumeSeries, candleData);
+            } else {
+                syncCrosshair(candleStickChart, candlestickSeries, volumeData);
+            }
+
+            handleDisplayLegend(candleData, volumeData);
+        };
+
+        candleStickChart.subscribeCrosshairMove(function (param) {
+            updateTitleData(param, 'candlestick');
+        });
+
+        volumeChart.subscribeCrosshairMove(function (param) {
+            updateTitleData(param, 'volume');
+        });
+
+        return () => {
+            candleStickChart.remove();
+            volumeChart.remove();
+            setLegendData({
+                O: '',
+                H: '',
+                L: '',
+                C: '',
+                VOLUME_KEY_NAME: '',
+                Change: '',
+                usdVol: '',
+            });
+            setHoverTime('');
+        };
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [data, setupCharts, timeScale]);
+
+    const titleOfChartClass =
+        'z-10 flex gap-4 text-2xs text-neutral-4 font-medium leading-4 pt-[0.375rem] px-4';
+    return (
+        <div className='bg-neutral-900 pt-[0.625rem]'>
+            <div className={clsx(titleOfChartClass)}>
+                {hoverTime && <span data-testid='hover-time'>{hoverTime}</span>}
+                {Object.entries(legendData)
+                    .filter(([key, _]) => legendArray.includes(key))
+                    .map(([key, value]) => {
+                        return (
+                            <div key={key} className='flex gap-1'>
+                                <span className='text-neutral-4'>{key}</span>
+                                <div
+                                    className={clsx('font-normal', {
```

Comment:
We should add the color to tailwind config instead of using the Hex here

Reply:
Made this change